internal/gps/pkgtree/pkgtree.go

```diff
@@ -921,3 +924,202 @@ func uniq(a []string) []string {
 	}
 	return a[:i]
 }
+
+// LibraryStatus represents one of a handful of possible statuses of a
+// particular subdirectory under vendor.
+type LibraryStatus uint8
```

Comment:
"library" is a bit of a loaded term - we avoid it throughout dep/gps in favor of the generic "project", specifically because we do not differentiate between dependencies that contain `package main`(s) and those that do not. 

given that, i think `VendorStatus` may be better here; this check is only meaningful in the context of a vendor directory, anyway.